301-subnet-driven-deployment/azuredeploy.json

```diff
@@ -0,0 +1,290 @@
+      },
+      "properties": {
+        "mode": "incremental",
+        "templateLink": {
+          "uri": "[concat(parameters('assetLocation'),'CreateADC.json')]",
+          "contentVersion": "1.0.0.1"
+        },
+        "parameters": {
+          "indx": { "value": "[copyIndex(1)]" },
+          "lbName": { "value": "[parameters('lbName')]" },
+          "publicIPAddressName": { "value": "[parameters('publicIPAddressName')]" },
+          "publicStartRdpPort": { "value": 5000 },
+          "vnetName": { "value": "[parameters('vnetName')]" },
+          "subnets": { "value": "[reference(parameters('vnetName')).outputs.vnetSubnets.value]" },
+          "storageAccountName": { "value": "[reference('LabStorage').outputs.storageAccountName.value]" },
+          "availabilitySetName": { "value": "AvSetDC" },
+          "vmNamePrefix": { "value": "dc" },
+          "nicNamePrefix": { "value": "nic" },
+          "domainName": { "value": "[parameters('domainName')]" },
+          "adminUserName": { "value": "[parameters('adminUserName')]" },
+          "adminPassword": { "value": "[parameters('adminPassword')]" },
+          "assetLocation": { "value": "[parameters('assetLocation')]" }
+        }
+      }
+    },
+    {
+      "apiVersion": "2015-01-01",
+      "name": "[concat('Members', copyIndex())]",
+      "type": "Microsoft.Resources/deployments",
+      "dependsOn": [
+        "Microsoft.Resources/deployments/RootFirstDC",
+        "Microsoft.Resources/deployments/setDNS"
+      ],
+      "copy": {
+        "name": "members",
+        "count": "[mul(length(variables('subnets')),2)]"
+      },
+      "properties": {
+        "mode": "Incremental",
+        "templateLink": {
+          "uri": "[concat(parameters('assetLocation'),'WinServ.json')]",
+          "contentVersion": "1.0.0.1"
+        },
+        "parameters": {
+          "indx": { "value": "[copyIndex()]" },
+          "lbName": { "value": "[parameters('lbName')]" },
+          "computerName": { "value": "srvMbmr" },
+          "publicStartRdpPort": { "value": 6000 },
+          "publicIPAddressName": { "value": "[parameters('publicIPAddressName')]" },
+          "vnetName": { "value": "[parameters('vnetName')]" },
+          "subnets": { "value": "[reference(parameters('vnetName')).outputs.vnetSubnets.value]" },
+          "storageAccountName": { "value": "[reference('LabStorage').outputs.storageAccountName.value]" },
+          "availabilitySetName": { "value": "AVSetDC" },
+          "domainName": { "value": "[parameters('domainName')]" },
+          "adminUsername": { "value": "[parameters('adminUserName')]" },
+          "adminPassword": { "value": "[parameters('adminPassword')]" },
+          "domainJoinUsername": { "value": "[parameters('adminUserName')]" },
+          "domainJoinPassword": { "value": "[parameters('adminPassword')]" },
+          "assetLocation": { "value": "[parameters('assetLocation')]" },
+          "dDisks": {
+            "value": [
```

Comment:
Yes, it is adjustable, by editing the array in the azuredeploy.json. The purpose is to avoid digging in the linked templates and keep everything that needs to be customized in the master template.

Currently there is very few array manipulation expressions which makes further parametrization quite difficult. Perhaps in a vNext?

I will edit the metadata.json asap.

Best regards,


  JF

From: Kay Singh [mailto:notifications@github.com]
Sent: Thursday, February 18, 2016 7:26 PM
To: Azure/azure-quickstart-templates <azure-quickstart-templates@noreply.github.com>
Cc: Jean-Francois Six <jean-francois.six@microsoft.com>
Subject: Re: [azure-quickstart-templates] 301 subnet driven deployment (#1184)


In 301-subnet-driven-deployment/azuredeploy.json<https://github.com/Azure/azure-quickstart-templates/pull/1184#discussion_r53357782>:

> +          "lbName": { "value": "[parameters('lbName')]" },

> +          "computerName": { "value": "srvMbmr" },

> +          "publicStartRdpPort": { "value": 6000 },

> +          "publicIPAddressName": { "value": "[parameters('publicIPAddressName')]" },

> +          "vnetName": { "value": "[parameters('vnetName')]" },

> +          "subnets": { "value": "[reference(parameters('vnetName')).outputs.vnetSubnets.value]" },

> +          "storageAccountName": { "value": "[reference('LabStorage').outputs.storageAccountName.value]" },

> +          "availabilitySetName": { "value": "AVSetDC" },

> +          "domainName": { "value": "[parameters('domainName')]" },

> +          "adminUsername": { "value": "[parameters('adminUserName')]" },

> +          "adminPassword": { "value": "[parameters('adminPassword')]" },

> +          "domainJoinUsername": { "value": "[parameters('adminUserName')]" },

> +          "domainJoinPassword": { "value": "[parameters('adminPassword')]" },

> +          "assetLocation": { "value": "[parameters('assetLocation')]" },

> +          "dDisks": {

> +            "value": [

Got it! It still one data disk per member right? From the readme I thought there was a way that a data disk per member would have been adjustable

—
Reply to this email directly or view it on GitHub<https://github.com/Azure/azure-quickstart-templates/pull/1184/files#r53357782>.